runelite-client/src/main/java/net/runelite/client/plugins/combat/CombatOverlay.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2018, Raqes <j.raqes@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.combat;
+
+import com.google.common.collect.ImmutableList;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.WorldType;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+
+@Slf4j
+public class CombatOverlay extends Overlay
+{
+	private static final Color TEXT_COLOR_BACKGROUND = new Color(0, 0, 0);
+	private static final Color TEXT_COLOR_FRONT = new Color(255, 255, 0);
+	private static final int TEXT_OFFSET = 10;
+	private static final int FIXED_VIEWPORT_OFFSET_X = 4;
+	private static final int FIXED_VIEWPORT_OFFSET_Y = 4;
+
+	private static final ImmutableList<WorldType> EXCLUDE_WORLD_TYPES = ImmutableList.of(
+		WorldType.DEADMAN,
+		WorldType.SEASONAL_DEADMAN,
+		WorldType.PVP,
+		WorldType.PVP_HIGH_RISK);
+
+	private Client client;
+
+	private CombatConfig config;
+
+	@Inject
+	CombatOverlay(Client client, CombatConfig config)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		this.client = client;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (isExcludedWorld() || !config.showCombatRange())
```

Comment:
The widget check is cheaper than this world check probably. I would do it first.